crates/db_views/src/private_message_view.rs

```diff
@@ -53,21 +53,27 @@ fn queries<'a>() -> Queries<
 
     // If its unread, I only want the ones to me
     if options.unread_only.unwrap_or(false) {
-      query = query
-        .filter(private_message::read.eq(false))
-        .filter(private_message::recipient_id.eq(recipient_id));
+      query = query.filter(private_message::read.eq(false));
+      if let Some(i) = options.creator_id {
+        query = query.filter(private_message::creator_id.eq(i))
+      } else {
+        query = query.filter(private_message::recipient_id.eq(recipient_id));
+      }
     }
     // Otherwise, I want the ALL view to show both sent and received
     else {
       query = query.filter(
         private_message::recipient_id
           .eq(recipient_id)
           .or(private_message::creator_id.eq(recipient_id)),
-      )
-    }
-
-    if let Some(i) = options.creator_id {
-      query = query.filter(private_message::creator_id.eq(i))
+      );
+      if let Some(i) = options.creator_id {
+        query = query.filter(
+          private_message::creator_id
+            .eq(i)
+            .or(private_message::recipient_id.eq(i)),
```

Comment:
As far as I can tell, this seems like its going to conflict with the condition on L68. IE L68 will show only messages that YOU sent, and filter out any others, so it won't even reach your condition.

Its hard to tell that it'll work correctly withouth unit tests.

Reply:
Ah, I see what you were thinking now, but no, line 68 ensures that the current user is *either* the creator or the recipient and 74 ensures the same condition for the specified user, the situation where they both are true is when both the current user and specified user are involved in some capacity